src/particles/elements/SoftSol.H

```diff
@@ -518,30 +522,31 @@ namespace SoftSolenoidData
             refpart.pt = pt;
 
             // push the linear map equations
-            amrex::Array2D<amrex::ParticleReal, 1, 6, 1, 6> const R = refpart.map;
+            amrex::SmallMatrix<amrex::ParticleReal, 6, 6> const R = refpart.map;
             amrex::ParticleReal const theta = tau*B0*bz/2.0_prt;
             amrex::ParticleReal const cs = std::cos(theta);
             amrex::ParticleReal const sn = std::sin(theta);
 
-            refpart.map(1,1) = R(1,1)*cs + R(3,1)*sn;
-            refpart.map(1,2) = R(1,2)*cs + R(3,2)*sn;
-            refpart.map(1,3) = R(1,3)*cs + R(3,3)*sn;
-            refpart.map(1,4) = R(1,4)*cs + R(3,4)*sn;
-
-            refpart.map(2,1) = R(2,1)*cs + R(4,1)*sn;
-            refpart.map(2,2) = R(2,2)*cs + R(4,2)*sn;
-            refpart.map(2,3) = R(2,3)*cs + R(4,3)*sn;
-            refpart.map(2,4) = R(2,4)*cs + R(4,4)*sn;
-
-            refpart.map(3,1) = R(3,1)*cs - R(1,1)*sn;
-            refpart.map(3,2) = R(3,2)*cs - R(1,2)*sn;
-            refpart.map(3,3) = R(3,3)*cs - R(1,3)*sn;
-            refpart.map(3,4) = R(3,4)*cs - R(1,4)*sn;
-
-            refpart.map(4,1) = R(4,1)*cs - R(2,1)*sn;
-            refpart.map(4,2) = R(4,2)*cs - R(2,2)*sn;
-            refpart.map(4,3) = R(4,3)*cs - R(2,3)*sn;
-            refpart.map(4,4) = R(4,4)*cs - R(2,4)*sn;
+            // note: map and R are zero-indexed
```

Comment:
Hm.  This zero-indexing is a major concern.  This indices 1-6 for the transfer matrix are standard and widely used, and this is very likely to lead to confusion (especially for me).  I am strongly in favor of using an object indexed 1-6.

Reply:
We could add a function to SmallMatrix that can be used to access the element using 1-based index. Not sure what to call it. But it could be something like `s.at<1>(i,j)` or a more explicit name.

Reply:
Or add an optional template parameter for starting index to SmallMatrix

Reply:
Fully agreed, thanks!

Reply:
done